File: 19-remove-nth-node-from-end-of-list/19-remove-nth-node-from-end-of-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        auto dummy = new ListNode();
        auto temp = head;
        dummy->next = head;
        // int k=1;
        while(n--){
            temp = temp->next;
            // k++;
        }
        while(temp){
            temp = temp->next;
            dummy = dummy->next;
        }
        
        cout<<dummy->val;
        if(dummy->next==head){
            return head->next;
        }else{
            dummy->next = dummy->next->next;
            return head;
        }

    }
};

File: 19-remove-nth-node-from-end-of-list/19-remove-nth-node-from-end-of-list.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def removeNthFromEnd(self, head, n):
        p1,p2 = head,head
        
        while n:
            p2 = p2.next
            n-=1
        
        if not p2:
            return head.next
        
        while p2.next:
            p2 = p2.next
            p1 = p1.next
        
        p1.next = p1.next.next
        return head
        